okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/OptionalMethod.java

```diff
@@ -169,6 +169,12 @@ private Method getMethod(Class<?> clazz) {
   private static Method getPublicMethod(Class<?> clazz, String methodName, Class[] parameterTypes) {
     Method method = null;
     try {
+      if (clazz == null) {
+        return null;
+      }
+      if ((clazz.getModifiers() & Modifier.PUBLIC) == 0) {
+        return getPublicMethod(clazz.getSuperclass(), methodName, parameterTypes);
+      }
```

Comment:
Sorry, could you explain what is going on more?  A private class with public methods should still be able to have the public methods called.  For example this works:

```
public interface Foo {
   void foo();
}
private class Bar implements Foo {
  @Override
  public void foo() {}
}

Foo f = new Bar();
f.foo();
```

Even though the class is itself private, public methods are still callable.  Can you link to the actual methods in conscrypt that were called?